owtf/webapp/src/containers/SettingsPage/index.js

```diff
@@ -2,19 +2,220 @@
+import { Button , Alert } from 'react-bootstrap';
+import {Grid, Col, Row, FormGroup, Form, ControlLabel, Nav, NavItem} from 'react-bootstrap';
+import { Tabs, Tab , TabContainer, TabContent, TabPane } from 'react-bootstrap';
 
-export default class SettingsPage extends React.Component {
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { makeSelectFetchError, makeSelectFetchLoading, makeSelectFetchConfigurations, makeSelectChangeError } from './selectors';
+import { loadConfigurations, changeConfigurations } from "./actions";
+import FormControl from "react-bootstrap/es/FormControl";
+import { ClipLoader } from 'react-spinners';
 
-  // Since state and props are static,
-  // there's no need to re-render this component
-  shouldComponentUpdate() {
-    return false;
+class SettingsPage extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.renderconfigurationTabsNav = this.renderconfigurationTabsNav.bind(this);
+    this.renderconfigurationTabsContent = this.renderconfigurationTabsContent.bind(this);
+    this.renderKeyDetails = this.renderKeyDetails.bind(this);
+    this.handleConfigurationChange = this.handleConfigurationChange.bind(this);
+    this.onUpdateConfiguration = this.onUpdateConfiguration.bind(this);
+    this.renderAlert = this.renderAlert.bind(this);
+    this.handleDismiss = this.handleDismiss.bind(this);
+    this.handleShow = this.handleShow.bind(this);
+
+    this.state = {
+      updateDisabled: true, //for update configuration button 
+      patch_data: {}, //contains information of the updated configurations
+      show: false, //handle alert visibility
+    };
+
+  }
+
+  handleDismiss() {
+    this.setState({ show: false });
+  }
+
+  handleShow() {
+    this.setState({ show: true });
+  }
+
+  //handles changes for all the configuration
+  handleConfigurationChange({ target }) {
+    this.setState({
+      patch_data: Object.assign({}, this.state.patch_data, {[target.name]: target.value}),
+      updateDisabled: false
+    });
+  }
+
+  //update the configurations using rest APIs
+  onUpdateConfiguration(){
+    this.props.onChangeConfiguration(this.state.patch_data);
+    this.setState({
+      patch_data: {},
+      updateDisabled: true,
+      show: true,
+    });
+    setTimeout(() => {this.setState({ show: false })}, 3000);
+  }
+
+  renderAlert(error){
+    if(this.state.show){
+      if(error !== false){
+        return (
+          <Alert bsStyle="danger" onDismiss={this.handleDismiss}>
+            {error.toString()}
+          </Alert>
+        )
+      }
+      else{
+        return(
+          <Alert bsStyle="success" onDismiss={this.handleDismiss}>
+            Confuguration updated successfully!
+          </Alert>
+        )
+      }
+    }
+  }
+
+  //Renders the configuratons tabs
+  renderconfigurationTabsNav(groupedConfigurations) {
+    let eventkey = 1;
```

Comment:
@sharmamohit123 You don't need to initialize a var to get loop index. `Map` function can get you that.
https://stackoverflow.com/questions/38364400/index-inside-map-function?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa